### PR TITLE
Ibp 5353 implement get breeding methods

### PR DESCRIPTION
**Pull request by lik-83**

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodDTO.java [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+import java.util.Date;
+import java.util.List;
+
 @AutoProperty
 public class BreedingMethodDTO extends BreedingMethodNewRequest {
 
 	private Integer mid;
 	private Boolean isBulkingMethod;
 	private Boolean isFavorite;
+	private Date date;
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodDTO.java:17]: @lik-83 could we use something more representative like creationDate?

File: src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java [hunk 2]
```diff
@@ -17,7 +17,8 @@
 
 	void delete(Integer breedingMethodDbId);
 
-	List<BreedingMethodDTO> getBreedingMethods(BreedingMethodSearchRequest methodSearchRequest, Pageable pageable);
+	List<BreedingMethodDTO> searchBreedingMethods(BreedingMethodSearchRequest methodSearchRequest, Pageable pageable,
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java:20]: remove final

File: src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java [hunk 3]
```diff
@@ -17,7 +17,8 @@
...
+			final String programUUID);
 
-	Long countBreedingMethods(BreedingMethodSearchRequest methodSearchRequest);
+	Long countSearchBreedingMethods(BreedingMethodSearchRequest methodSearchRequest, final String programUUID);
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java:23]: @lik-83 Remove final

File: src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodDTO.java (hunk 1 above)
Comment (lik-83) [reply src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodDTO.java]: Done

File: src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java (hunk 2 above)
Comment (lik-83) [reply src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java:20]: Done

File: src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java (hunk 3 above)
Comment (lik-83) [reply src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodService.java]: Done

**Pull request merged**